fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java

```diff
@@ -169,6 +171,26 @@ private void addBackend(String host, int heartbeatPort, boolean isFree, String d
         MetricRepo.generateBackendsTabletMetrics();
     }
 
+    public void updateBackendAddress(UpdateBackendAddressClause updateBackendAddressClause) throws DdlException {
```

Comment:
I think this operation require a write lock when updating catalog but I can't find where it does.

Reply:
Yes, it does needs a lock to prevent data mistakes, but i don't think we should solve the lock problem in this PR, we should push a new PR to solve it. otherwise, it would make this PR code confusing

Reply:
I think if it is a bug, fixing in this PR is reasonable because the basic logic is relevant. After all, this is your PR, I suggest you ask for a second opinion.

Reply:
The method `process`, that calls this method is a synchronized method. So it's not a problem